artichoke-backend/src/extn/core/kernel/require.rs

```diff
@@ -162,4 +166,65 @@ pub mod method {
         }
         Err(Error::CannotLoad(args.file.to_owned()))
     }
+
+    fn load_impl(interp: &Artichoke, args: Args, base: &str) -> Result<Require, Error> {
+        let interp = interp.clone();
+        let mut path = PathBuf::from(args.file);
+        let files = if path.is_relative() {
+            path = PathBuf::from(base);
+            let mut files = Vec::with_capacity(2);
+            if !args.file.ends_with(".rb") {
+                files.push(path.join(format!("{}.rb", args.file)))
+            }
+            files.push(path.join(args.file));
+            files
+        } else {
+            vec![path.join(args.file)]
+        };
+
+        for path in files {
+            let is_file = {
+                let api = interp.0.borrow();
+                api.vfs.is_file(path.as_path())
+            };
+
+            if !is_file {
+                continue;
+            }
+
+            let metadata = {
+                let api = interp.0.borrow();
+                api.vfs.metadata(path.as_path()).unwrap_or_default()
+            };
+            let file = if let Some(filename) = path.as_path().to_str() {
+                filename
+            } else {
+                "(require)"
+            };
+
+            let contents = {
+                let api = interp.0.borrow();
+                api.vfs.read_file(path.as_path())
+            };
+
+            let require = Require {
+                file: file.to_owned(),
+                rust: metadata.require,
+                ruby: contents.ok(),
+            };
+            let borrow = interp.0.borrow();
+            borrow
+                .vfs
+                .set_metadata(path.as_path(), metadata)
```

Comment:
we can skip setting the metadata since we're not changing it.

In `Kernel#require`, we set the metadata to mark that a file has been required. Since `Kernel#load` doesn't alter that state, we do not need to write metadata back into the VFS.

Reply:
Removed.

Reply:
Removed